File: binary_search_algo/dart_bin_search/recursive_method/lib/recursive_method.dart
binarySearch(int target, List<int> array, int high, int low) {
  // Base case, in case the target is not in the array
  if (low > high) return -1;

  int middle = (low + (high - low) / 2).toInt();
  print(middle);

  if (target == array[middle]) return middle;
  if (target < array[middle]) return binarySearch(target, array, middle - 1, low);
  if (target > array[middle]) return binarySearch(target, array, high, middle + 1);
}

File: binary_search_algo/js_bin_search/recursive_method.js

function main() {
    const array = [1,3,5,7,9,11,13]
    var target = 5
    let low = 0;
    let high = array.length - 1;
    let result;
    result = binarySearch(target, array, low, high)

    if (result == -1) {
        console.log("The element was not found")
    } else {
        console.log("The result is: " + result)
    }
}

function binarySearch(target, array, low, high) {
    if (low<=high) {
       let middle = low + Math.floor((high - low) / 2);
       let arrayValue = array[middle]
       if(target==arrayValue) 
            return middle
       if(target<arrayValue) 
            binarySearch(target, array, low, middle -1)
       if(target>arrayValue)
            binarySearch(target, array, middle + 1, high) 
    }
    // Base case
    return -1
}

main()